internal/scan-manager/scanners/common/src/main/java/org/wso2/security/tools/scanmanager/scanners/common/ScannerController.java

```diff
@@ -57,26 +64,50 @@ public ScannerController(Scanner scanner) throws IOException {
     /**
      * Start a new scan.
      *
-     * @param scanRequest Object that represent the required information for the scanner operation
+     * @param scannerScanRequest Object that represent the required information for the scanner operation
      * @return whether the start scan request is accepted
      */
     @PostMapping("scan")
     @ResponseBody
-    public ResponseEntity startScan(@RequestBody ScannerScanRequest scanRequest) {
+    public ResponseEntity startScan(@RequestBody ScannerScanRequest scannerScanRequest) {
         ResponseEntity responseEntity;
+        responseEntity = validateStartScanReq(scannerScanRequest);
+        if (responseEntity.getStatusCode().equals(HttpStatus.ACCEPTED)) {
+            if (scanner.validateStartScan(scannerScanRequest)) {
+                log.info("Invoking start scan API.");
+                startScanThread = new Thread(() -> scanner.startScan(scannerScanRequest), "StartScanThread");
 
-        if (!hasScanStarted) {
-            log.info("Invoking start scan API.");
-            responseEntity = scanner.startScan(scanRequest);
-            if (responseEntity.getStatusCode().equals(HttpStatus.ACCEPTED)) {
+                startScanThread.start();
                 hasScanStarted = true;
+            } else {
+                String message = "Start scan request validation is failed.";
+                responseEntity = new ResponseEntity<>(new ErrorMessage(HttpStatus.BAD_REQUEST.value(),
+                        message), HttpStatus.BAD_REQUEST);
+            }
+        }
+        return responseEntity;
+    }
+
+    private ResponseEntity validateStartScanReq(ScannerScanRequest scannerScanRequest) {
+        ResponseEntity responseEntity;
+        if (!hasScanStarted) {
+            if (!StringUtils.isEmpty(scannerScanRequest.getAppId())) {
+                if (!StringUtils.isEmpty(scannerScanRequest.getJobId())) {
+                    responseEntity = new ResponseEntity<>(HttpStatus.ACCEPTED);
+                } else {
+                    String message = "Job Id is missing in the request.";
+                    responseEntity = new ResponseEntity<>(new ErrorMessage(HttpStatus.BAD_REQUEST.value(),
+                            message), HttpStatus.BAD_REQUEST);
+                }
+            } else {
+                String message = "Application Id is missing in the request.";
+                responseEntity = new ResponseEntity<>(new ErrorMessage(HttpStatus.BAD_REQUEST.value(),
+                        message), HttpStatus.BAD_REQUEST);
             }
         } else {
             String message = "Cannot start a new scan since another scan is in progress.";
```

Comment:
Shouldn't we log this event too? Same applies to other error conditions where we only response the requests without doing logging.